src/test/mockmodelinstancechangingstates.hpp

```diff
@@ -42,9 +42,10 @@ class MockModelInstanceChangingStates : public ovms::ModelInstance {
         status.setAvailable();
         return ovms::StatusCode::OK;
     }
-    void unloadModel(bool isPermanent = true) override {
-        status.setUnloading();
-        status.setEnd();
+    void unloadModel(bool isPermanent = true, bool isError = false) override {
+        ovms::ModelVersionStatusErrorCode errorStatus = isError ? ovms::ModelVersionStatusErrorCode::ERROR : ovms::ModelVersionStatusErrorCode::OK;
+        status.setUnloading(errorStatus);
+        status.setEnd(errorStatus);
```

Comment:
Why we would want to have model with status END(retired) with error code? It would suggest that something went wrong during requested retiring, why not have loading state with error?